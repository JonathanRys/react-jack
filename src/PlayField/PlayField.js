import React, { Component } from "react";
import PropTypes from "prop-types";

import { Container, Row } from "reactstrap";

import "./play-field.css";

import Dealer from "./Dealer/Dealer";
import Deck from "./Deck/Deck";
import Player from "./Player/Player";
import Badge from "./Badge/Badge";

export default class PlayField extends Component {
  render() {
    const player = this.props.player;
    const dealer = this.props.dealer;
    const index = player.handIndex;
    return (
      <Container className="PlayField_main">
        <Row>
          <Badge
            name={dealer.name}
            avatar={dealer.avatar}
            score={dealer.score}
            dealer={true}
            flipped={!this.props.turn.playersTurn}
          />
          <Dealer
            hand={dealer.hand}
            flipped={!this.props.turn.playersTurn}
            drawnCard={
              this.props.turn.playersTurn ? null : this.props.deck.drawnCard
            }
            takeCard={this.props.giveDealerCard}
          />
        </Row>
        <Row>
          <Deck />{" "}
          {/* Really just an image I guess.  Maybe add some animation... */}
        </Row>
        <Row>
          <Player
            playersTurn={this.props.turn.playersTurn}
            hands={player.hands}
            drawnCard={
              this.props.turn.playersTurn === player.playerIndex
                ? this.props.deck.drawnCard
                : null
            }
            takeCard={this.props.giveCard}
          />
          <Badge
            name={player.name}
            avatar={player.avatar}
            score={player.score[index]}
            balance={player.balance}
            currentBet={player.currentBet}
            setBet={this.props.setBet}
            dealer={false}
            flipped={true}
          />
        </Row>
      </Container>
    );
  }
}

PlayField.propTypes = {
  player: PropTypes.object.isRequired,
  dealer: PropTypes.object.isRequired,
  deck: PropTypes.object.isRequired,
  turn: PropTypes.object.isRequired
};
