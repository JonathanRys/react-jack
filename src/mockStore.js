export default {
  dealer: {
    name: "Dealer",
    avatar: "./images/dealer.png",
    hand: ["HA", "C5"],
    score: 0,
    busted: false,
    hasBlackjack: false
  },
  player: {
    name: "Player1",
    avatar: "./images/default.png",
    playerIndex: 1,

    balance: 500,
    currentBet: 5,

    hands: [["SA", "CK"], ["D2", "HJ"]],
    status: [null],
    score: [0],
    busted: [false],
    hasBlackjack: [false],
    hasInsurance: [false],
    splitHand: false,
    playerStands: [false]
  },
  turn: {
    isPlaying: false,
    numPlayers: 2,
    playersTurn: 1,
    error: false,
    errMsg: ""
  },
  deck: {
    drawnCard: null,
    deck: []
  },
  subscribe: jest.fn(),
  dispatch: jest.fn(),
  getState: jest.fn()
};
