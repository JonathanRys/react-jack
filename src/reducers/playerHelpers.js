export default function getScore(state) {
  // Define scoring enum
  const points = {
    "2": 2,
    "3": 3,
    "4": 4,
    "5": 5,
    "6": 6,
    "7": 7,
    "8": 8,
    "9": 9,
    "10": 10,
    J: 10,
    Q: 10,
    K: 10,
    A: 11
  };

  // Declare a function to return the score
  const scoreHand = cards => {
    // Set initial values
    let score = 0;
    let numberOfAces = 0;

    // Calculate score based on points enum
    score = cards
      .map(card => {
        if (card.slice(1) === "A") numberOfAces++;
        return points[card.slice(1)];
      })
      .reduce((a, b) => a + b);

    // Score aces as 1 if they bring the score over 21
    while (numberOfAces && score > 21) {
      score -= 10;
      --numberOfAces;
    }

    return score;
  };

  // Score the current hand
  const index = state.handIndex;
  const newScore = scoreHand(state.hands[index]);

  // Return a new object with the updated score
  return {
    score: replaceAtIndex(state.score, newScore, index),
    busted: replaceAtIndex(state.busted, newScore > 21, index),
    hasBlackjack: replaceAtIndex(
      state.hasBlackjack,
      newScore === 21 && state.hands[index].length === 2,
      index
    )
  };
}

export function replaceAtIndex(parentArray, itemToAdd, index) {
  return [
    ...parentArray.slice(0, index),
    itemToAdd,
    ...parentArray.slice(index + 1)
  ];
}

export function addCard(hands, card, index = 0) {
  const newHand = [...hands[index], card];

  return {
    hands: replaceAtIndex(hands, newHand, index)
  };
}
