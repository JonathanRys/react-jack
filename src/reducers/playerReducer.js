import getScore, { addCard } from "./playerHelpers";

const initialState = {
  name: "Player1",
  avatar: "./images/default.png",
  playerIndex: 1,
  handIndex: 0,

  balance: 495,
  currentBet: 5,

  hands: [[]],
  status: [null],
  score: [0],
  busted: [false],
  hasBlackjack: [false],
  hasInsurance: [false],
  splitHand: false,
  playerStands: [false]
};

const INSURANCE_RATE = 0.5;

export default function playerReducer(state = initialState, action) {
  const index = state.handIndex;
  switch (action.type) {
    case "SET_NAME":
      return { ...state, name: action.payload.name };
    case "SET_AVATAR":
      return { ...state, avatar: action.payload.avatar };
    case "SET_STATUS":
      return {
        ...state,
        status: [
          ...state.status.slice(0, index),
          action.payload.status,
          ...state.status.slice(index + 1)
        ]
      };
    case "TAKE_CARD":
      const newCards = addCard(state.hands, action.payload.card, index);
      return {
        ...state,
        ...getScore({
          ...state,
          ...newCards
        }),
        ...newCards
      };
    case "CLEAR_HANDS":
      return {
        ...initialState,
        name: state.name,
        avatar: state.avatar,
        balance: state.balance,
        currentBet: state.currentBet
      };
    case "STAND":
      return {
        ...state,
        playerStands: [
          ...state.playerStands.slice(0, index),
          true,
          ...state.playerStands.slice(index + 1)
        ]
      };
    case "BUY_CHIPS":
      return { ...state, balance: state.balance + action.payload.newChips };
    case "SET_BET":
      return { ...state, currentBet: action.payload.newBet };
    case "LOSE_BET":
      return { ...state, balance: state.balance - state.currentBet };
    case "WIN_BET":
      const multiplier = action.payload ? action.payload.multiplier || 1 : 1;
      return {
        ...state,
        balance: state.balance + state.currentBet * multiplier
      };
    case "SET_INSURED":
      return {
        ...state,
        balance: state.balance - state.currentBet * INSURANCE_RATE,
        hasInsurance: [
          ...state.hasInsurance.slice(0, index),
          true,
          ...state.hasInsurance.slice(index + 1)
        ]
      };
    case "CREDIT":
      return { ...state, balance: state.balance + action.payload.delta };
    case "DEBIT":
      return { ...state, balance: state.balance - action.payload.delta };
    default:
      return state;
  }
}
