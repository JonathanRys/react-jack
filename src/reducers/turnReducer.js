const initialState = {
  isPlaying: false,
  numPlayers: 2,
  playersTurn: 1,
  error: false,
  errMsg: ""
};

export default function turnReducer(state = initialState, action) {
  switch (action.type) {
    case "PLAY":
      return { ...initialState, isPlaying: true };
    case "STOP":
      return { ...initialState, isPlaying: false };
    case "DEALER_TURN":
      return { ...state, playersTurn: 0 };
    case "ADD_PLAYER":
      return {
        ...state,
        numPlayers: state.numPlayers < 9 ? state.numPlayers + 1 : 9
      };
    case "NEXT_PLAYER":
      return {
        ...state,
        playersTurn: (state.playersTurn + 1) % state.numPlayers
      };
    default:
      return state;
  }
}
