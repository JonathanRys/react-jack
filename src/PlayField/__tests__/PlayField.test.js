import React from "react";
import { shallow } from "enzyme";

import PlayField from "../PlayField";

import mockStore from "../../mockStore";
import { dealerAddCard } from "../../reducers/dealerHelpers";

const mockProps = {
  ...mockStore,
  giveCard: jest.fn(),
  giveDealerCard: jest.fn(),
  clearCard: jest.fn()
};

describe("Test PlayField component", () => {
  it("matches the snapshot", () => {
    const testProps = { ...mockProps };
    const component = shallow(<PlayField {...testProps} />);
    expect(component).toMatchSnapshot();
  });

  it("tests playersTurn", () => {
    const testProps = { ...mockProps };
    testProps.turn.playersTurn = 1;
    const component = shallow(<PlayField {...testProps} />);
    expect(component).toMatchSnapshot();
  });
});
