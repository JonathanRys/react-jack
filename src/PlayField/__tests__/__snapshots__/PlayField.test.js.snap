// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PlayField component matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlayField
    clearCard={[MockFunction]}
    dealer={
      Object {
        "avatar": "./images/dealer.png",
        "busted": false,
        "hand": Array [
          "HA",
          "C5",
        ],
        "hasBlackjack": false,
        "name": "Dealer",
        "score": 0,
      }
    }
    deck={
      Object {
        "deck": Array [],
        "drawnCard": null,
      }
    }
    dispatch={[MockFunction]}
    getState={[MockFunction]}
    giveCard={[MockFunction]}
    giveDealerCard={[MockFunction]}
    player={
      Object {
        "avatar": "./images/default.png",
        "balance": 500,
        "busted": Array [
          false,
        ],
        "currentBet": 5,
        "hands": Array [
          Array [
            "SA",
            "CK",
          ],
          Array [
            "D2",
            "HJ",
          ],
        ],
        "hasBlackjack": Array [
          false,
        ],
        "hasInsurance": Array [
          false,
        ],
        "name": "Player1",
        "playerIndex": 1,
        "playerStands": Array [
          false,
        ],
        "score": Array [
          0,
        ],
        "splitHand": false,
      }
    }
    subscribe={[MockFunction]}
    turn={
      Object {
        "errMsg": "",
        "error": false,
        "isPlaying": true,
        "numPlayers": 2,
        "playersTurn": 0,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Row
          tag="div"
        >
          <Badge
            avatar="./images/dealer.png"
            dealer={true}
            flipped={true}
            name="Dealer"
            score={0}
          />
          <Dealer
            drawnCard={null}
            flipped={true}
            hand={
              Array [
                "HA",
                "C5",
              ]
            }
            takeCard={[MockFunction]}
          />
        </Row>,
        <Row
          tag="div"
        >
          <Deck />
           
        </Row>,
        <Row
          tag="div"
        >
          <Player
            drawnCard={null}
            hands={
              Array [
                Array [
                  "SA",
                  "CK",
                ],
                Array [
                  "D2",
                  "HJ",
                ],
              ]
            }
            playersTurn={0}
            takeCard={[MockFunction]}
          />
          <Badge
            avatar="./images/default.png"
            balance={500}
            currentBet={5}
            dealer={false}
            flipped={true}
            name="Player1"
            score={undefined}
            setBet={undefined}
          />
        </Row>,
      ],
      "className": "PlayField_main",
      "tag": "div",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Badge
              avatar="./images/dealer.png"
              dealer={true}
              flipped={true}
              name="Dealer"
              score={0}
            />,
            <Dealer
              drawnCard={null}
              flipped={true}
              hand={
                Array [
                  "HA",
                  "C5",
                ]
              }
              takeCard={[MockFunction]}
            />,
          ],
          "tag": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "avatar": "./images/dealer.png",
              "dealer": true,
              "flipped": true,
              "name": "Dealer",
              "score": 0,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "drawnCard": null,
              "flipped": true,
              "hand": Array [
                "HA",
                "C5",
              ],
              "takeCard": [MockFunction],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Deck />,
            " ",
          ],
          "tag": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          " ",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Player
              drawnCard={null}
              hands={
                Array [
                  Array [
                    "SA",
                    "CK",
                  ],
                  Array [
                    "D2",
                    "HJ",
                  ],
                ]
              }
              playersTurn={0}
              takeCard={[MockFunction]}
            />,
            <Badge
              avatar="./images/default.png"
              balance={500}
              currentBet={5}
              dealer={false}
              flipped={true}
              name="Player1"
              score={undefined}
              setBet={undefined}
            />,
          ],
          "tag": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "drawnCard": null,
              "hands": Array [
                Array [
                  "SA",
                  "CK",
                ],
                Array [
                  "D2",
                  "HJ",
                ],
              ],
              "playersTurn": 0,
              "takeCard": [MockFunction],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "avatar": "./images/default.png",
              "balance": 500,
              "currentBet": 5,
              "dealer": false,
              "flipped": true,
              "name": "Player1",
              "score": undefined,
              "setBet": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Row
            tag="div"
          >
            <Badge
              avatar="./images/dealer.png"
              dealer={true}
              flipped={true}
              name="Dealer"
              score={0}
            />
            <Dealer
              drawnCard={null}
              flipped={true}
              hand={
                Array [
                  "HA",
                  "C5",
                ]
              }
              takeCard={[MockFunction]}
            />
          </Row>,
          <Row
            tag="div"
          >
            <Deck />
             
          </Row>,
          <Row
            tag="div"
          >
            <Player
              drawnCard={null}
              hands={
                Array [
                  Array [
                    "SA",
                    "CK",
                  ],
                  Array [
                    "D2",
                    "HJ",
                  ],
                ]
              }
              playersTurn={0}
              takeCard={[MockFunction]}
            />
            <Badge
              avatar="./images/default.png"
              balance={500}
              currentBet={5}
              dealer={false}
              flipped={true}
              name="Player1"
              score={undefined}
              setBet={undefined}
            />
          </Row>,
        ],
        "className": "PlayField_main",
        "tag": "div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Badge
                avatar="./images/dealer.png"
                dealer={true}
                flipped={true}
                name="Dealer"
                score={0}
              />,
              <Dealer
                drawnCard={null}
                flipped={true}
                hand={
                  Array [
                    "HA",
                    "C5",
                  ]
                }
                takeCard={[MockFunction]}
              />,
            ],
            "tag": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "avatar": "./images/dealer.png",
                "dealer": true,
                "flipped": true,
                "name": "Dealer",
                "score": 0,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "drawnCard": null,
                "flipped": true,
                "hand": Array [
                  "HA",
                  "C5",
                ],
                "takeCard": [MockFunction],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Deck />,
              " ",
            ],
            "tag": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            " ",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Player
                drawnCard={null}
                hands={
                  Array [
                    Array [
                      "SA",
                      "CK",
                    ],
                    Array [
                      "D2",
                      "HJ",
                    ],
                  ]
                }
                playersTurn={0}
                takeCard={[MockFunction]}
              />,
              <Badge
                avatar="./images/default.png"
                balance={500}
                currentBet={5}
                dealer={false}
                flipped={true}
                name="Player1"
                score={undefined}
                setBet={undefined}
              />,
            ],
            "tag": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "drawnCard": null,
                "hands": Array [
                  Array [
                    "SA",
                    "CK",
                  ],
                  Array [
                    "D2",
                    "HJ",
                  ],
                ],
                "playersTurn": 0,
                "takeCard": [MockFunction],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "avatar": "./images/default.png",
                "balance": 500,
                "currentBet": 5,
                "dealer": false,
                "flipped": true,
                "name": "Player1",
                "score": undefined,
                "setBet": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Test PlayField component tests playersTurn 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlayField
    clearCard={[MockFunction]}
    dealer={
      Object {
        "avatar": "./images/dealer.png",
        "busted": false,
        "hand": Array [
          "HA",
          "C5",
        ],
        "hasBlackjack": false,
        "name": "Dealer",
        "score": 0,
      }
    }
    deck={
      Object {
        "deck": Array [],
        "drawnCard": null,
      }
    }
    dispatch={[MockFunction]}
    getState={[MockFunction]}
    giveCard={[MockFunction]}
    giveDealerCard={[MockFunction]}
    player={
      Object {
        "avatar": "./images/default.png",
        "balance": 500,
        "busted": Array [
          false,
        ],
        "currentBet": 5,
        "hands": Array [
          Array [
            "SA",
            "CK",
          ],
          Array [
            "D2",
            "HJ",
          ],
        ],
        "hasBlackjack": Array [
          false,
        ],
        "hasInsurance": Array [
          false,
        ],
        "name": "Player1",
        "playerIndex": 1,
        "playerStands": Array [
          false,
        ],
        "score": Array [
          0,
        ],
        "splitHand": false,
      }
    }
    subscribe={[MockFunction]}
    turn={
      Object {
        "errMsg": "",
        "error": false,
        "isPlaying": true,
        "numPlayers": 2,
        "playersTurn": 1,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Row
          tag="div"
        >
          <Badge
            avatar="./images/dealer.png"
            dealer={true}
            flipped={false}
            name="Dealer"
            score={0}
          />
          <Dealer
            drawnCard={null}
            flipped={false}
            hand={
              Array [
                "HA",
                "C5",
              ]
            }
            takeCard={[MockFunction]}
          />
        </Row>,
        <Row
          tag="div"
        >
          <Deck />
           
        </Row>,
        <Row
          tag="div"
        >
          <Player
            drawnCard={null}
            hands={
              Array [
                Array [
                  "SA",
                  "CK",
                ],
                Array [
                  "D2",
                  "HJ",
                ],
              ]
            }
            playersTurn={1}
            takeCard={[MockFunction]}
          />
          <Badge
            avatar="./images/default.png"
            balance={500}
            currentBet={5}
            dealer={false}
            flipped={true}
            name="Player1"
            score={undefined}
            setBet={undefined}
          />
        </Row>,
      ],
      "className": "PlayField_main",
      "tag": "div",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Badge
              avatar="./images/dealer.png"
              dealer={true}
              flipped={false}
              name="Dealer"
              score={0}
            />,
            <Dealer
              drawnCard={null}
              flipped={false}
              hand={
                Array [
                  "HA",
                  "C5",
                ]
              }
              takeCard={[MockFunction]}
            />,
          ],
          "tag": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "avatar": "./images/dealer.png",
              "dealer": true,
              "flipped": false,
              "name": "Dealer",
              "score": 0,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "drawnCard": null,
              "flipped": false,
              "hand": Array [
                "HA",
                "C5",
              ],
              "takeCard": [MockFunction],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Deck />,
            " ",
          ],
          "tag": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          " ",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Player
              drawnCard={null}
              hands={
                Array [
                  Array [
                    "SA",
                    "CK",
                  ],
                  Array [
                    "D2",
                    "HJ",
                  ],
                ]
              }
              playersTurn={1}
              takeCard={[MockFunction]}
            />,
            <Badge
              avatar="./images/default.png"
              balance={500}
              currentBet={5}
              dealer={false}
              flipped={true}
              name="Player1"
              score={undefined}
              setBet={undefined}
            />,
          ],
          "tag": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "drawnCard": null,
              "hands": Array [
                Array [
                  "SA",
                  "CK",
                ],
                Array [
                  "D2",
                  "HJ",
                ],
              ],
              "playersTurn": 1,
              "takeCard": [MockFunction],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "avatar": "./images/default.png",
              "balance": 500,
              "currentBet": 5,
              "dealer": false,
              "flipped": true,
              "name": "Player1",
              "score": undefined,
              "setBet": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Row
            tag="div"
          >
            <Badge
              avatar="./images/dealer.png"
              dealer={true}
              flipped={false}
              name="Dealer"
              score={0}
            />
            <Dealer
              drawnCard={null}
              flipped={false}
              hand={
                Array [
                  "HA",
                  "C5",
                ]
              }
              takeCard={[MockFunction]}
            />
          </Row>,
          <Row
            tag="div"
          >
            <Deck />
             
          </Row>,
          <Row
            tag="div"
          >
            <Player
              drawnCard={null}
              hands={
                Array [
                  Array [
                    "SA",
                    "CK",
                  ],
                  Array [
                    "D2",
                    "HJ",
                  ],
                ]
              }
              playersTurn={1}
              takeCard={[MockFunction]}
            />
            <Badge
              avatar="./images/default.png"
              balance={500}
              currentBet={5}
              dealer={false}
              flipped={true}
              name="Player1"
              score={undefined}
              setBet={undefined}
            />
          </Row>,
        ],
        "className": "PlayField_main",
        "tag": "div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Badge
                avatar="./images/dealer.png"
                dealer={true}
                flipped={false}
                name="Dealer"
                score={0}
              />,
              <Dealer
                drawnCard={null}
                flipped={false}
                hand={
                  Array [
                    "HA",
                    "C5",
                  ]
                }
                takeCard={[MockFunction]}
              />,
            ],
            "tag": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "avatar": "./images/dealer.png",
                "dealer": true,
                "flipped": false,
                "name": "Dealer",
                "score": 0,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "drawnCard": null,
                "flipped": false,
                "hand": Array [
                  "HA",
                  "C5",
                ],
                "takeCard": [MockFunction],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Deck />,
              " ",
            ],
            "tag": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            " ",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Player
                drawnCard={null}
                hands={
                  Array [
                    Array [
                      "SA",
                      "CK",
                    ],
                    Array [
                      "D2",
                      "HJ",
                    ],
                  ]
                }
                playersTurn={1}
                takeCard={[MockFunction]}
              />,
              <Badge
                avatar="./images/default.png"
                balance={500}
                currentBet={5}
                dealer={false}
                flipped={true}
                name="Player1"
                score={undefined}
                setBet={undefined}
              />,
            ],
            "tag": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "drawnCard": null,
                "hands": Array [
                  Array [
                    "SA",
                    "CK",
                  ],
                  Array [
                    "D2",
                    "HJ",
                  ],
                ],
                "playersTurn": 1,
                "takeCard": [MockFunction],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "avatar": "./images/default.png",
                "balance": 500,
                "currentBet": 5,
                "dealer": false,
                "flipped": true,
                "name": "Player1",
                "score": undefined,
                "setBet": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
